sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -380,6 +380,9 @@ object ColumnPruning extends Rule[LogicalPlan] {
         p
       }
 
+    // Eliminate the Projects with empty projectList
+    case p @ Project(projectList, child) if projectList.isEmpty => child
```

Comment:
Actually, I just try to run this tests without this patch. It passes. @gatorsmile Can you verify it?